### PR TITLE
Remove schedule name uniqueness restriction

### PR DESCRIPTION
**Pull request by matiasb**
Related to https://github.com/grafana/oncall/issues/1452

### CODE REVIEW

**Review by mderynck — APPROVED** [review]

File: engine/apps/schedules/migrations/0012_auto_20230502_1259.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('schedules', '0011_oncallschedule_cached_ical_final_schedule'),
+    ]
+
+    operations = [
+        migrations.AlterModelOptions(
```
Comment (vadimkerr) [inline engine/apps/schedules/migrations/0012_auto_20230502_1259.py:13]: Looks like nothing changed about the model manager, is this expected?

**Review by vadimkerr — APPROVED** [review]
LGTM 👍 
nit: maybe add a simple unit test creating two schedules with the same name?

File: engine/apps/schedules/migrations/0012_auto_20230502_1259.py (hunk 1 above)
Comment (matiasb) [reply engine/apps/schedules/migrations/0012_auto_20230502_1259.py:13]: This was added by the automatic migration generation, will double-check.

Comment (matiasb) [reply engine/apps/schedules/migrations/0012_auto_20230502_1259.py:13]: This seems related to a previous change in the manager, for which there wasn't a migration created. I think the change was adding a method to the custom base queryset, which is used as the default `objects` manager via `PolymorphicManager` (so, not a real model/db change, but it seems this is detected as a modification for which a migration step is added).

**Pull request merged**